File: scripts/functions-lua/kp.lua

Kp = {}

local function getch_unix()
	os.execute("stty cbreak </dev/tty >/dev/tty 2>&1")
	local key = io.read(1)
	return(key);
end

Kp.get_id_process = function(process)
	if process then
		local cmd = 'pgrep '..process
		local all_process = {}
		local file = io.popen(cmd)
		for v in file:lines() do
			table.insert(all_process, v)
		end
		return all_process
	end
end

Kp.check_args = function(process)
	if process then return true else return false end
end

Kp.kill_process = function(process)
	os.execute('kill '..process)
end

Kp.get_input = function(process)
	print('You want kill ALL process from '.. process..'?(y/n)')
	local key_pressed = getch_unix()
	print()
	return key_pressed
end

Kp.confirmate = function(key_pressed)
	return key_pressed == 'y'
end

return Kp

File: scripts/python/kp.py
#!/usr/bin/env python

import subprocess
import argparse
import os



# get arg
def getArg():
    parser = argparse.ArgumentParser()
    parser.add_argument('program', help='Program to kill')
    args = parser.parse_args()
    return args.program

def getProcess(process, list_process):
    p = subprocess.Popen(['ps', '-A'], stdout=subprocess.PIPE)
    out, err = p.communicate()
    for lines in out.splitlines():
        if process.encode() in lines:
            list_process.append(lines)

def killProcess(listProcess):
    for process in listProcess:
        pid = int(process.split(None, 1)[0])
        os.kill(pid, 9)

def getInput():
    keyPressed = input('You want kill ALL process from ' + process + '?(y/n)\n')
    return keyPressed

def confirmate(key):
    return key == 'y'


list_process = []
process = getArg()

getProcess(process, list_process)

keyPressed = getInput()

if confirmate(keyPressed):
    killProcess(list_process)
